docker/docker_transport.go

```diff
@@ -0,0 +1,120 @@
+package docker
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/types"
+	distreference "github.com/docker/distribution/reference"
+)
+
+// Transport is an ImageTransport for Docker references.
+var Transport = dockerTransport{}
+
+type dockerTransport struct{}
+
+func (t dockerTransport) Name() string {
+	return "docker"
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an ImageReference.
+func (t dockerTransport) ParseReference(reference string) (types.ImageReference, error) {
+	return ParseReference(reference)
+}
+
+// dockerReference is an ImageReference for Docker images.
+type dockerReference struct {
+	ref reference.Named
+	tag string
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an Docker ImageReference.
+func ParseReference(reference string) (types.ImageReference, error) {
+	return parseReference(reference)
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an dockerReference.
+func parseReference(refString string) (dockerReference, error) {
```

Comment:
Three things called parseReference in a single file, _and_ a `NewReference` to boot. Feels a bit silly. Yes, it would make callers easier to read, not sure that it is worth it.